app/messaging/process-message-request.js

```diff
@@ -4,13 +4,16 @@ import { validateMessageRequest } from './message-request-schema.js'
 export const processMessageRequest = async (logger, message, receiver) => {
   try {
     logger.warn(`Received sfd message request with id: ${message.messageId}`)
-    const messageBody = message.body
-    if (validateMessageRequest(logger, messageBody)) {
-      await sendMessageToSingleFrontDoor(logger, message.messageId, messageBody)
-      await receiver.completeMessage(message)
+
+    if (!validateMessageRequest(logger, message.body)) {
+      await receiver.deadLetterMessage(message) // not worth retrying, invalid input
+      return
     }
```

Comment:
Coding guidelines encourage using the principle of "separate in order to name" in situations like this, over using comments. https://defra.github.io/software-development-standards/standards/common_coding_standards/

Intent could perhaps be better expressed with a named value for the validationResponse. (Although personally I think the code is pretty clear as is). Something along the lines of:

```
const retryMaySucceed = validateMessageRequest(logger, messageBody)

if (retryMaySucceed) {
  ...
}
```

Reply:
Thanks Matt, will update

Reply:
Agreed, code updated